src/components/generic/InputField/InputField.jsx

```diff
@@ -13,57 +13,61 @@ const HINTS = {
 };
 
 /**
- * Input Field component.
+ * Generic input field component for building more specific components (text, date, number, etc.).
+ * Should not be used directly in forms.
  *
  * @param {object} props  Data passed to the component
+ * @param {string} [props.ariaLabel]  Aria label
  * @param {boolean} [props.autoFocus=false]  Flag indicating if the input should have autoFocus
  * @param {string} [props.className]  Custom CSS class to apply
- * @param {object} [props.style]  Custom style to apply
- * @param {string} [props.type=text]  Input type
- * @param {string} [props.name]  Input name
- * @param {(string|number)} [props.value]  Input value
- * @param {string} [props.placeholder]  Input placeholder
- * @param {number} [props.max]  Maximum value for the input element
- * @param {number} [props.min]  Minimum value for the input element
- * @param {number} [props.maxLength]  Maximum number of characters allowed
- * @param {string} [props.pattern]  Specifies a regular expression that the element's value is checked against
- * @param {Function} [props.onChange]  Action to perform on input change
  * @param {boolean} [props.disabled=false]  Flag indicating input disabled
  * @param {string} [props.error]  Error text
  * @param {string} [props.label]  Label text
- * @param {string} [props.ariaLabel]  Hint to be displayed as aria-label
+ * @param {number} [props.max]  Maximum value for the input element
+ * @param {number} [props.maxLength]  Maximum number of characters allowed
+ * @param {number} [props.min]  Minimum value for the input element
+ * @param {string} [props.name]  Input name
+ * @param {Function} [props.onChange]  Action to perform on input change
+ * @param {string} [props.pattern]  Specifies a regular expression that the element's value is checked against
+ * @param {string} [props.placeholder]  Input placeholder
  * @param {boolean} [props.required=false]  Flag indicating input required
+ * @param {(boolean|object)} [props.rightIcon]  Icon to be displayed on the right side of the input
```

Comment:
{boolean|object}

Reply:
better yet, {false|React.JSX.Element}

Reply:
to avoid the warning I put: {boolean|React.JSX.Element}